spec/models/manageiq/providers/openstack/infra_manager/refresher_rhos_juno_spec.rb

```diff
@@ -44,7 +44,7 @@
       EmsRefresh.refresh(@ems.network_manager)
       @ems.reload
 
-      @host = ManageIQ::Providers::Openstack::InfraManager::Host.all.detect { |x| x.name.include?('(NovaCompute)') }
+      @host = ManageIQ::Providers::Openstack::InfraManager::Host.all.order(:ems_ref).detect { |x| x.name.include?('(NovaCompute)') }
```

Comment:
Ah, the name of the host is its openstack id appended with its 'role', determined by how it is deployed for the overcloud.  I'm guessing that when we re-recorded VCRs we create a deployment with two compute nodes, leading to this issue.